src/screencast/screencast.c

```diff
@@ -1,61 +1,113 @@
+#define _POSIX_C_SOURCE 200809L
+
 #include "screencast.h"
 
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <sys/mman.h>
+#include <spa/utils/result.h>
+
+#include "pipewire_screencast.h"
+#include "wlr_screencast.h"
+#include "xdpw.h"
+#include "logger.h"
+
 static const char object_path[] = "/org/freedesktop/portal/desktop";
 static const char interface_name[] = "org.freedesktop.impl.portal.ScreenCast";
 
-int setup_outputs(struct xdpw_state *state) {
+void xdpw_screencast_instance_init(struct xdpw_screencast_context *ctx,
+		struct xdpw_screencast_instance *cast, struct xdpw_wlr_output *out, bool with_cursor){
+	cast->ctx = ctx;
+	cast->target_output = out;
+	cast->framerate = out->framerate;
+	cast->with_cursor = with_cursor;
+	cast->refcount = 1;
+	logprint(INFO, "xdpw: screencast instance %p has %d references", cast, cast->refcount);
+	wl_list_insert(&ctx->screencast_instances, &cast->link);
+	logprint(INFO, "xdpw: %d active screencast instances",
+		wl_list_length(&ctx->screencast_instances));
+}
+
+void xdpw_screencast_instance_destroy(struct xdpw_screencast_instance *cast){
+	logprint(TRACE, "xdpw: destroying cast instance");
```

Comment:
Should this be called only when `cast->refcount == 0`? If so, would be nice to add an assertion.